### ISSUE TITLE
Exclude special (or defined by user) tokens from the activations

### ISSUE THREAD

**Issue by avgalichin** [EleutherAI/sparsify#104]
In some cases (e.g. training on the residual stream | on activations from the Instruct model) it can be useful to exclude activations corresponding to special tokens when training. We can also add support for user-defined tokens to exclude (with the parameter similar to the one in [saelens](https://github.com/jbloomAus/SAELens/blob/main/sae_lens/config.py#L143)).

Looking at the code, this could be done without big modifications:

1. From the config, instantiate the `torch.Tensor` with tokens to exclude inside the `Trainer.__init__`:
```python
self.exclude_special_tokens = torch.tensor(
    self.cfg.exclude_special_tokens, device=device, dtype=torch.long
)
```
2. Inside `fit`, instantiate the dynamic mask alongside the metric definitions:
```python
# set the dummy mask for the 0-th iteration
batch = next(iter(dl))
x = batch["input_ids"].to(device)
train_tokens_mask = torch.ones_like(x, dtype=torch.bool).flatten(0, 1)

 acc_steps = self.cfg.grad_acc_steps * self.cfg.micro_acc_steps
 denom = acc_steps * self.cfg.wandb_log_frequency
 num_tokens_in_step = 0

avg_auxk_loss = defaultdict(float)
...
3. Use the mask inside the hook as an outer scope variable to exclude activations. 
```python
def hook(...):
...
outputs = outputs.flatten(0, 1)
inputs = inputs.flatten(0, 1) if self.cfg.sae.transcode else outputs

# Remove tokens not used for training
all_outputs = outputs.detach().clone()  # NOTE: required for ce/kl loss
outputs = outputs[train_tokens_mask]
inputs = inputs[train_tokens_mask]
...
if self.cfg.loss_fn in ("ce", "kl"):
    # Replace the normal output with the SAE output
    output = torch.where(train_tokens_mask.unsqueeze(-1), out.sae_out, all_outputs)  # sets grads of excluded tokens to zero
    output = output.reshape(out_shape).type_as(outputs)
```
4. For each newly received batch, set the corresponding mask:
```python
for batch in dl:
    x = batch["input_ids"].to(device)
        train_tokens_mask = torch.isin(x, self.exclude_tokens, invert=True) \
             .flatten(0, 1)
   ...
   N = train_tokens_mask.sum().item()
```

**Comment by avgalichin** [comment]
If this is okay, I can work on it in the way proposed above or whichever approach you think is better.
